app/modules/ads1115.c

```diff
@@ -513,6 +495,16 @@ static int ads1115_lua_read(lua_State *L) {
     return 3;
 }
 
+static int ads1115_lua_delete(lua_State *L) {
+    ads_ctrl_ud_t *ads_ctrl = luaL_checkudata(L, 1, metatable_name);
+    if (ads_ctrl->timer_ref != LUA_NOREF) {
+        os_timer_disarm(&ads_ctrl->timer);
+        lua_rawgeti(L, LUA_REGISTRYINDEX, ads_ctrl->timer_ref);
```

Comment:
Why push timer_ref onto the stack? It's redundant at least and might leak memory since the return below says 0 results (not sure, can't run the code without a chip).

Reply:
A bit of cargo on my side - I took a snipped out of readoutdone without fully understanding what it does. Fixing it explained some other anomaly I have seen.